orchestrators/FinanceOperations/src/Services/DepreciationRunService.php

```
@@ -233,8 +245,33 @@ private function calculateDepreciation(array $assetData): float
+                        $periodId,
+                        'Sum-of-years depreciation requires a positive sum-of-years denominator'
+                    );
+                }
+
+                $depreciationBase = $depreciableAmount / $years;
+                if ($depreciationBase == 0.0) {
+                    throw DepreciationCoordinationException::runFailed(
+                        $tenantId,
+                        $periodId,
+                        'Sum-of-years depreciation requires a non-zero depreciation base'
+                    );
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\Nexus\FinanceOperations\Exceptions\DepreciationCoordinationException' in method 'calculateDepreciation'. (undefined)

(StaticAccess)